tests/DriverManagerTest.php

```diff
@@ -113,6 +114,7 @@ public function testDatabaseUrlPrimaryReplica(): void
             'wrapperClass' => PrimaryReadReplicaConnection::class,
         ];
 
+        $this->expectDeprecationWithIdentifier('https://github.com/doctrine/dbal/pull/5843');
```

Comment:
URLs for primary/replica have to be parsed in advance as well, so we do get the deprecation here.